### PR TITLE
Fast Mode for MakeNNVars

### PR DESCRIPTION
**Pull request by cros0400**
Taking from Josh's version of MakeNNVars analyzer, this will allow us to use fast mode for making the NN variables. Also, includes a few changes to represent the correct variables per channel

### CODE REVIEW

File: Analyzer/src/MakeNNVariables.cc [hunk 1]
```diff
@@ -166,145 +179,140 @@ void MakeNNVariables::Loop(NTupleReader& tr, double, int maxevents, bool)
...
-                std::string ptCut = "pt30";
-                
-                std::set<std::string> varChannelSpecific =
-                {
-                    "HT_trigger_"+ptCut+myVarSuffix,
-                    "NGoodJets_"+ptCut+"_double"+myVarSuffix,
-                };
+                        std::set<std::string> var7toLastJet =
+                        {
+                            "combined7thToLastJet_pt_cm"+myVarSuffix,
```
Comment (JHiltbrand) [inline Analyzer/src/MakeNNVariables.cc:250]: Is this the one we want for all channels ? I figured we would also need a 6thToLast and 8thToLast

File: Analyzer/test/Plotters/General/stackPlotter_aux_nnInputs.py [hunk 2]
```diff
@@ -29,7 +31,7 @@
     #"h_Jet@_cm_CSVb_?"        : {"logY" : True,  "orders" : list(xrange(1,8)), "Y" : {"title" : "A.U.", "min" : 0.2}, "X" : {"title" : "Jet @ DeepCSV b",   "rebin" : 1, "min" :  0,  "max" :    1}},
```
Comment (JHiltbrand) [inline Analyzer/test/Plotters/General/stackPlotter_aux_nnInputs.py:31]: Can we knock out these "old" CSV guys at this point ?

File: Analyzer/test/Plotters/General/stackPlotter_aux_nnInputs.py [hunk 3]
```diff
@@ -50,7 +52,7 @@
 
 backgrounds = {
     "TT"       : {"name" : "t#bar{t} + jets", "color" : 40,  "lstyle" : 1, "mstyle" : 8, "lsize" : 3, "msize" : 0},
-    "QCD"      : {"name" : "QCD multijet",    "color" : 30,  "lstyle" : 1, "mstyle" : 8, "lsize" : 3, "msize" : 0},
+    #"QCD"      : {"name" : "QCD multijet",    "color" : 30,  "lstyle" : 1, "mstyle" : 8, "lsize" : 3, "msize" : 0},
```
Comment (JHiltbrand) [inline Analyzer/test/Plotters/General/stackPlotter_aux_nnInputs.py:55]: Should be kept in for the long run

**Comment by JHiltbrand** [comment]
Hey @cros0400,

I assume you produced the latest NN inputs with this branch ?
If so, I guess this branch can be merged in at some point ?
Or at least you just have to get in your latest changes and merge (I see the branch is in conflict with `master`)

**Comment by cmadrid1** [comment]
Hey @cros0400 is there gonna be any progress on the PR or should it be closed?

**Comment by cros0400** [comment]
Sorry @JHiltbrand and @cmadrid1, I should have responded earlier. I think these changes got dragged into my latest PR with updates to all of Josh's comments. That was the version of the NN input analyzer that I used for the latest set.

**Pull request merged**